library/init/util.lean

```diff
@@ -49,3 +49,6 @@ meta def undefined {α : Sort u} : α := undefined_core "undefined"
 
 meta def unchecked_cast {α : Sort u} {β : Sort u} : α → β :=
 cast undefined
+
+/-- For tactics to tag proofs -/
+@[reducible] def id_tag {α : Sort u} (tag : string) (x : α) : α := x
```

Comment:
```suggestion
@[reducible] def id_tag (tag : string) (x : Prop) : Prop := x
```

AFAICT it's not used for anything else and we'd like to avoid unnecessary functions in data.

Using strings for data certainly has some performance impact (particularly when every tactic inserts a string into the proof term).  I'd be a bit more comfortable if it was `(tag : unit)` and we had `def ring := ()`, etc.  Or even `theorem ring_tag {p : Prop} (h : p) : p := h`.

The macro is also an option, but that requires adapting the various export formats.